src/pybind/mgr/cephadm/module.py

```diff
@@ -1186,13 +1187,17 @@ def add_host(self, spec):
         :param host: host name
         """
         assert_valid_host(spec.hostname)
-        out, err, code = self._run_cephadm(spec.hostname, 'client', 'check-host',
-                                           ['--expect-hostname', spec.hostname],
-                                           addr=spec.addr,
-                                           error_ok=True, no_fsid=True)
-        if code:
-            raise OrchestratorError('New host %s (%s) failed check: %s' % (
-                spec.hostname, spec.addr, err))
+        try:
+            out, err, code = self._run_cephadm(spec.hostname, 'client', 'check-host',
+                                               ['--expect-hostname', spec.hostname],
+                                               addr=spec.addr,
+                                               error_ok=True, no_fsid=True)
+            if code:
+                raise OrchestratorError('New host %s (%s) failed check: %s' % (
+                    spec.hostname, spec.addr, err))
+        except execnet.gateway_bootstrap.HostNotFound as e:
```

Comment:
also, occurs when the `ceph.pub` ssh key is not distributed...which is somewhat misleading.

Would be nice if we could catch a more general exception, but it appears `execnet` will only raise `HostNotFound` for all these conditions.